modules/geo-layers/src/tile-layer/tile-layer.js

```diff
@@ -96,6 +98,8 @@ export default class TileLayer extends CompositeLayer {
       this.state.tileset.tiles.forEach(tile => {
         tile.layers = null;
       });
+    } else if (changeFlags.viewportChanged && tileset.isLoaded) {
```

Comment:
This does not look right. `tileset.isLoaded` is updated by `this._updateTileset()`. Before then it's only reflecting the last viewport state.

Reply:
For example, while you are zooming in on the map `_updateTileset` set `tileset.isLoaded` to false because new tiles have started to download. This condition prevent that  onViewportChange be called if you drag or zoom map while tiles are downloading

Reply:
Correct, except `_updateTileset` has not been called yet when you check this flag.

Reply:
mmm, we could change the order.

```
this._updateTileset();
if (changeFlags.viewportChanged && tileset.isLoaded) {
 this._onViewportChange();
}
```

but when you start to dag the map `tileset.isLoaded` is still true it will continue to be true until you've daged enough to ask for new tiles. Honestly, I can't think of how to do it without a debounce function

Reply:
In any case, I'll think about it a little more tomorrow